### DISCUSSION TITLE
Problems Installing Iroh?

### DISCUSSION THREAD

**Discussion by b5** [n0-computer/iroh#1591] [General]
have an issue installing iroh? Post about it here!

**Comment by mattdesl** [comment]
Following the docs today:

```
curl -fsSL https://sh.iroh.computer/install.sh | sh
```

Result:

```
Fetching iroh from https://vorc.s3.us-east-2.amazonaws.com/iroh-darwin-aarch64-latest
######################################################################## 100.0%
mv: rename /Users/matt/Library/Application Support/iroh/bin/iroh to /usr/local/bin/iroh: Permission denied
mv: fastcopy: open() failed (to): /usr/bin/iroh: Operation not permitted
Moved iroh to /Users/matt/.local/bin/iroh

iroh was installed successfully to /Users/matt/.local/bin/iroh
Run 'iroh --help' to get started

Stuck? Check out this discussion: https://github.com/n0-computer/iroh/discussions/1591
```

**Reply by ramfox** [reply]
looks like it installed successfully, did you attempt to run `iroh` and get errors?

**Reply by mattdesl** [reply]
I ended up saving the script to a file and running it with sudo (couldn’t get it to run without that error otherwise). It works now.

**Reply by dignifiedquire** [reply]
```
mv: rename /Users/matt/Library/Application Support/iroh/bin/iroh to /usr/local/bin/iroh: Permission denied
```

Yeah we try copying it into the users bin directory, which it seems your user wasn't allowed to do

**Comment by 2color** [comment]
After upgrading Iroh from `0.10` to `0.18`, I tried running `iroh start` and got the following error message:

```
➜  ~ iroh start
Error: Table 'download-policy-1' does not exist
```

Any pointers as to why that's happening?

**Reply by dignifiedquire** [reply]
we don't support database migrations between all versions, you can either install versions in between to get the migration, or reset your iroh directory

**Reply by 2color** [reply]
Where is the default iroh directory?

**Reply by matheus23** [reply]
Depends on the platform:
| platform | directory | example |
|-- | -- | -- |
| Linux	| `$XDG_DATA_HOME` or `$HOME/.local/share`	| `/home/alice/.local/share/iroh` |
| macOS |	`$HOME/Library/Application Support`	| `/Users/Alice/Library/Application Support/iroh` |
| Windows |	`{FOLDERID_RoamingAppData}`	| `C:\Users\Alice\AppData\Roaming/iroh` |

It can also be set via an env var `IROH_DATA_DIR`.

We should document this. I'm personally a fan of a command in CLIs like `iroh paths` that just tells me where stuff is.
